### ISSUE TITLE
Add Top button for easily reaching the top page

### ISSUE THREAD

**Issue by mshwf** [flutter/website#1950]
Page URL: https://docs.flutter.dev/get-started/flutter-for/xamarin-forms-devs
Page source: https://github.com/flutter/website/tree/main/src/get-started/flutter-for/xamarin-forms-devs.md

Description:
Please add Top button, it's extremely important for long web pages, when reading from cell phones. Instead of running my fingers on the screen.

**Comment by chalin** [comment]
@galeyang @InMatrix - I've been increasingly testing the site from mobile as well, and I've felt this need too, especially because several site pages are very long. Any feedback concerning this?

As you might recall, I had implemented (via #1271) a "scroll to top" button based on the webdev site implementation, but we decided against it because it wasn't in the original design (cf. #1272).

IIRC Daniel had suggested that if we choose implement such a feature, it should be a floating FAB in the lower right corner.

cc @maguinis @kwalrath

**Comment by InMatrix** [comment]
Sounds reasonable to me. It would be mostly for Android phone users though, since on iOS you could just tap the top of the screen. It would be nice if we can avoid showing this button on Mobile Safari.

**Comment by galeyang** [comment]
Agreed with this issue. The proposal of floating button in lower right corner sounds good because we won't able to see fixed table of content on a narrow screen, like mobile.

**Comment by sfshaza2** [comment]
I like this idea, though I'm not sure @chalin has time for this before he leaves.

**Comment by InMatrix** [comment]
@galeyang and I are assigned to the issue, but I think fixing this issue requires a developer. I will unassigned Gale and I for now. Hope we will have resource to add this simple feature on mobile soon.

**Comment by parlough** [comment]
As part of the new mobile version of the table of contents ("On this page") implemented in https://github.com/flutter/website/issues/11750, you can now quickly navigate to the top of the page (or any other header) with a button in the opened dialog. 

<img width="397" alt="Mobile TOC opened in layout tutorial" src="https://github.com/user-attachments/assets/2d56d1a5-1abb-4fe5-a117-8d1b23c4a3f9" />

Thanks for the feedback and for all the discussion here.
